airflow/models/pool.py

```diff
@@ -48,6 +50,45 @@ def get_pool(pool_name, session=None):
     def get_default_pool(session=None):
         return Pool.get_pool(Pool.DEFAULT_POOL_NAME, session=session)
 
+    @staticmethod
+    @provide_session
+    def slots_stats(session=None) -> Dict[str, Dict[str, int]]:
+        from airflow.models.taskinstance import TaskInstance  # Avoid circular import
+
+        pools = {}
+
+        pool_rows = session.query(Pool.pool, Pool.slots).all()
+        for (pool_name, total_slots) in pool_rows:
+            pools[pool_name] = {
+                "total": total_slots,
+                "used": 0,
```

Comment:
How about
```suggestion
                State.RUNNING: 0,
``` 

etc?

Reply:
I personally would prefer this, but didn't make that change to make it backwards compatible. Are you cool with making this breaking change?

Reply:
Ohhhh yeah it changes the metrics doesn't it :(

Reply:
Wait, not strictly, but it's probably a bit odd to have different names here and in the metric that is emitted.

Reply:
cool, I will add this breaking change then.

Reply:
What do you think about using a typed object instead of a dictionary, e.g. NamedTuple? In Python 3.8, we have TypedDict, but we still support Python 3.6+: https://www.python.org/dev/peps/pep-0589/

Reply:
good suggestion, updated to use TypedDict @mik-laj